src/borrow_tracker/tree_borrows/wildcard.rs

```diff
@@ -0,0 +1,592 @@
+                        // None -> Write
+                        // Read -> Write
+                        parent_state.child_writes += 1;
+                    }
+                    if old_child_access == None {
+                        // None -> Read
+                        // None -> Write
+                        parent_state.child_reads += 1;
+                    }
+                } else {
+                    if old_child_access == Write {
+                        // Write -> None
+                        // Write -> Read
+                        parent_state.child_writes -= 1;
+                    }
+                    if new_child_access == None {
+                        // Read  -> None
+                        // Write -> None
+                        parent_state.child_reads -= 1;
+                    }
+                }
+
+                let old_parent_local_access = old_parent_state.max_local_access();
+                let new_parent_local_access = parent_state.max_local_access();
+
+                {
+                    // We need to update the `max_foreign_access` of `child`'s siblings.
+                    // For this we can reuse the `push_relevant_children` function.
+                    //
+                    // We pass it just the siblings without child itself. Since `child`'s
+                    // `max_local_access()` is foreign to all of its siblings we can pass
+                    // it as part of the foreign access.
+
+                    // it doesnt matter wether we read these from old or new
+                    let constant_factors =
+                        max(parent_state.exposed_as, parent_state.max_foreign_access);
+
+                    // `state` contains the correct child_writes/reads counts for just the
+                    // siblings excluding `child`.
+                    let state = if is_upgrade { &old_parent_state } else { parent_state };
+
+                    Self::push_relevant_children(
+                        &mut stack,
+                        max(constant_factors, new_child_access),
+                        max(constant_factors, old_child_access),
+                        state.child_reads,
+                        state.child_writes,
+                        parent_node.children.iter().copied().filter(|id| child != *id),
+                        wildcard_accesses,
+                    );
+                }
+                if old_parent_local_access == new_parent_local_access {
+                    // We didn't change `max_local_access()` for parent, so we don't need to propagate further upwards.
+                    break;
+                }
+
+                old_child_access = old_parent_local_access;
+                new_child_access = new_parent_local_access;
+                child = parent_id;
+            }
+        }
+        // Traverses up the tree to update max_foreign_access fields of children and cousins who need to be updated.
```

Comment:
This goes down the tree, not up, doesn't it?

Remember, the root is at the top.